themes/morphosource/views/MyProjects/Dashboard/manage_all_download_requests_html.php

```diff
@@ -1,50 +1,202 @@
+				$va_pending_download_request['requested_on'], 
+				array('dateFormat' => 'delimited', 'timeOmit' => true));
 			if($va_pending_download_request['request']){
-				print "<br/>".$va_pending_download_request['request'];
+				print "<br/><b>Request text:</b> ".$va_pending_download_request['request'];
 			}
+
+			print "</div>";
+
+			print "<div class='' style='display: inline-block; vertical-align: middle;'>";
+			$t_media->load($va_pending_download_request['media_id']);
+			print "<div style='border: 1px solid #e8e8e8; margin-top: 15px; margin-right: 15px;'>".$t_media->getPreviewMediaFile(null, array('icon', 'thumbnail'))['media']['thumbnail']."</div>";
+			print "</div>";
+
 			print "</div>\n";
+
+			print "<div style='text-align: center; margin-top: 20px; margin-bottom: 5px;'><a href='#' class='button buttonMedium' onclick='msMediaDownloadRequestDashboardApprove({$va_pending_download_request['request_id']}); return false;'>"._t('Approve')."</a>&nbsp;&nbsp;&nbsp;<a href='#' class='button buttonMedium' onclick='msMediaDownloadRequestDashboardDeny({$va_pending_download_request['request_id']}); return false;'>"._t('Deny')."</a></div>";
+
+			print "</div>";
+			print "</div>"; 
+			print "</div>";  
 		}
+		print "</form>";
 	}else{
-		if($this->getVar("approval_success")){
-			print "<H2><b>"._t("All pending requests have been approved.")."</b></H2>";
-		}
 		print "<p>There are no pending download requests associated with any of your projects.</p>";
 	}
 ?>
 	</div>
+
 	<script type="text/javascript">
 		function msMediaDownloadRequestDashboardApprove(request_id) {
 			jQuery('#msMediaDownloadRequestContainer').load('<?php print caNavUrl($this->request, 'MyProjects', 'Dashboard', 'ApproveDownloadRequest'); ?>/request_id/' + request_id + '/manage_all/1');
 		}
 		function msMediaDownloadRequestDashboardDeny(request_id) {
 			jQuery('#msMediaDownloadRequestContainer').load('<?php print caNavUrl($this->request, 'MyProjects', 'Dashboard', 'DenyDownloadRequest'); ?>/request_id/' + request_id + '/manage_all/1');
 		}
+
+		jQuery('#allCheck').change(function () {
+			if (this.checked) {
+				jQuery('input').prop('checked', true);
+			} else {
+				jQuery('input').prop('checked', false);
+			}
+		})
+
+		jQuery('.projectCheck').change(function () {
+			if (this.checked) {
+				jQuery('#'+this.id+'.projectItem').find('input').prop('checked', true);
+				jQuery('[id*="' + this.id + '"].projectUserCheck').prop('checked', true);
+			} else {
+				jQuery('#'+this.id+'.projectItem').find('input').prop('checked', false);
+				jQuery('[id*="' + this.id + '"].projectUserCheck').prop('checked', false);
+				jQuery('#allCheck').prop('checked', false);
+			}
+		})
+
+		jQuery('.projectUserCheck').change(function () {
+			var id_combo = this.id;
+			var id_array = id_combo.split('-');
+			if (this.checked) {
+				jQuery('#'+id_array[0]+'.projectItem').children('#'+id_array[1]+'.requestUserItem').find('input').prop('checked', true);
```

Comment:
this is fine if the element id always has two dashes.  just to make sure :)

Reply:
The element id in these cases is always formatted "[project id number]-[user id number]", so it doesn't have two dashes, but it will still always produce the expected two-element array.